v8/quotebacks/quotebacks_mdx.py

```diff
@@ -0,0 +1,240 @@
+    """
+
+    def __init__(self, md, config):
+        """ Usually a tree processor won't take a config dict, but here
+        we're overriding init to accept one. This is for future options.
+
+        Stash the markdown instance for later.
+        """
+        self.config = config
+        self.md = md
+
+        # We record whether there were quotebacks found
+        self.md.quotebacks_found = False
+
+        super().__init__(md)
+
+    def run(self, root):
+        # Iterate over all blockquote elements
+        for bq in root.iter("blockquote"):
+            # blockquote must have >1 children
+            if len(bq) <= 1:
+                LOGGER.info("blockquote must have >1 children")
+                continue
+
+            # blockquote's final child must be a paragraph
+            if bq[-1].tag != "p":
+                LOGGER.info("blockquote's final child must be a p tag")
+                continue
+
+            # Keep the paragraph element handy. It must look like
+            # '-- AUTHOR <a href="CITE_URL">TITLE</a>'
+            p_elem = bq[-1]
+
+            if not (len(p_elem) == 1 and p_elem[0].tag == "a"):
+                LOGGER.info("Final p must have one child and it must be an a tag")
+                continue
+
+            a_elem = p_elem[0]
+            title = a_elem.text
+            cite_url = a_elem.get("href", None)
+
+            if not title:
+                LOGGER.info("Final a tag must have text for the title")
+                continue
+
+            if not cite_url:
+                LOGGER.info("Final a tag must have an href for the cite URL")
+                continue
+
+            # There must be no tail text
+            if p_elem.tail is not None:
+                LOGGER.info("p must have no tail text")
+                continue
+
+            # The start text must follow a strict pattern
+            if not p_elem.text:
+                LOGGER.info("p must have start text")
+
+            m = re.match(r"^-- (?P<author>.+),\s+$", p_elem.text)
+            if not m:
+                LOGGER.info("p must match the pattern '-- AUTHOR, '")
+                continue
+
+            author = m.groupdict()["author"]
+
+            # Time to change the document!
+            # First, remove the original element
+            bq.remove(p_elem)
+
+            # Build the replacement tag, using the original elements where possible
+            footer = ET.SubElement(bq, "footer",)
+            footer.text = p_elem.text
+            cite = ET.SubElement(footer, "cite")
+            cite.append(a_elem)
+
+            # Set bq Attributes
+            attrib = {
+                "class": "quoteback",
+                "data-title": title,
+                "data-author": author,
+                "cite": cite_url,
+            }
+
+            [bq.set(k, v) for k, v in attrib.items()]
+
+            LOGGER.info("Successful: blockquote -> quoteback")
+            self.md.quotebacks_found = True
+
+
+def test():
```

Comment:
This isn’t relevant to Nikola, perhaps remove it.

Reply:
Done.